plainly-aescripts/src/project.ts

```diff
@@ -0,0 +1,73 @@
+const propNameId = 'xmp:PlainlyProjectId';
+const propNameRevision = 'xmp:PlainlyProjectRevision';
+const propNameName = 'xmp:PlainlyProjectName';
+
+function setProjectData(id: string, revision: string, name: string) {
+  const project = app.project;
+  project.save();
+
+  if (ExternalObject.AdobeXMPScript === undefined) {
+    // load the XMPlibrary as an ExtendScript ExternalObject
+    ExternalObject.AdobeXMPScript = new ExternalObject('lib:AdobeXMPScript');
+  }
+
+  const mdata = new XMPMeta!(project.xmpPacket); //get the project's XMPmetadata
+  // update the Label project metadata's value
+  const schemaNS = XMPMeta!.getNamespaceURI('xmp');
+  try {
+    mdata.setProperty(schemaNS, propNameId, id);
+    mdata.setProperty(schemaNS, propNameRevision, revision);
+    mdata.setProperty(schemaNS, propNameName, name);
+    project.xmpPacket = mdata.serialize();
+  } catch (e) {
+    return `Error: ${e}`;
+  } finally {
+    project.save();
+  }
+}
+
+function getProjectData() {
+  const project = app.project;
+
+  if (ExternalObject.AdobeXMPScript === undefined) {
+    // load the XMPlibrary as an ExtendScript ExternalObject
+    ExternalObject.AdobeXMPScript = new ExternalObject('lib:AdobeXMPScript');
+  }
+
+  const mdata = new XMPMeta!(project.xmpPacket); //get the project's XMPmetadata
+  const schemaNS = XMPMeta!.getNamespaceURI('xmp');
+
+  const projectData = {
+    id: String(mdata.getProperty(schemaNS, propNameId)),
+    revision: String(mdata.getProperty(schemaNS, propNameRevision)),
+    name: String(mdata.getProperty(schemaNS, propNameName)),
+  };
+
+  if (projectData.id === 'undefined') {
```

Comment:
* this is a bit confusing why would we get the `undefined` string back?
* should we simply do this check earlier simply:  
  ```
  const projectId = mdata.getProperty(schemaNS, propNameId);
  if (!projectId) {
    return undefined;
  }
  ```
* or maybe meta has some helper `hasProperty()`?